### DISCUSSION TITLE
Can anyone help me run these commands when starting i3wm without asking for password? Arch linux

### DISCUSSION THREAD

**Discussion by ronibastos** [i3/i3#6291] [Q&A] [answered]
vim .config/i3/config

exec --no-startup-id alacritty -e sudo pacman -Scc --noconfirm
exec --no-startup-id alacritty -e sudo pacman -Syyuu --noconfirm

Can anyone help me run these commands when starting i3wm without asking for password? Arch linux

**Comment by yusufaktepe** [comment, marked as answer]
> alacritty -e sudo pacman -Syyuu --noconfirm

You really shouldn't run upgrades unattended on Arch.

That being said, you can do this by adding those commands to the sudoers NOPASSWD list.

Create a new sudoers file (`sudo visudo /etc/sudoers.d/10-nopasswd`) with the following content:

```sudoers
%wheel ALL=(ALL) NOPASSWD:SETENV: \
  /usr/bin/pacman -Scc --noconfirm,\
  /usr/bin/pacman -Syyuu --noconfirm
```

Run with i3:

```
exec --no-startup-id alacritty -e sh -c "sudo pacman -Scc --noconfirm; sudo pacman -Syyuu --noconfirm"
```

**Reply by ronibastos** [reply]
Thank you very much! Everything went well!!

**Comment by ronibastos** [comment]
Thank you very much! Everything went well!!


 

    Em terça-feira, 5 de novembro de 2024 às 15:01:00 BRT, Yusuf Aktepe ***@***.***> escreveu:  
 
 



alacritty -e sudo pacman -Syyuu --noconfirm


You really shouldn't run upgrades unattended on Arch.

That being said, you can do this by adding those commands to the sudoers NOPASSWD list.

Create a new sudoers file (sudo visudo /etc/sudoers.d/10-nopasswd) with the following content:
%wheel ALL=(ALL) NOPASSWD:SETENV: \
  /usr/bin/pacman -Scc --noconfirm,\
  /usr/bin/pacman -Syyuu --noconfirm

Run with i3:
exec --no-startup-id alacritty -e sh -c "sudo pacman -Scc --noconfirm; sudo pacman -Syyuu --noconfirm"
